manual/parts/vspace.tex

```diff
@@ -257,11 +255,11 @@ \subsection{ASID Control}
 
 \subsection{ASID Pool}
 
-An \obj{ASID Pool} confers the right to create a subset of the available
-maximum applications. For a VSpace to be usable by an application, it
-must be assigned to an ASID. This is done using a capability to an
-\obj{ASID Pool}. The \obj{ASID Pool} object has a single method, \texttt{Assign}, for each
-architecture:
+An \obj{ASID Pool} confers the right to use a subset of the globally available address space
+identifiers. The size of this subset is architecture dependent. For a VSpace object to be usable by
+a thread, it must be assigned to an ASID via an \obj{ASID Pool} capability. Each ASID can be
+assigned to at most one VSpace. The \obj{ASID Pool} capability has a single invocation,
+\texttt{Assign}, for each architecture.
```

Comment:
Is there a good reason why assigning a vspace to an asid pool doesn't require you to specify the slot index to use? The kernel can spend a long time searching for a free asid slot if it's always trying to find the next free one. And it's easy to reuse ASIDs unintentionally if you revoke an address space and then create a new one, it'll use the same slot I believe.

Reply:
> Is there a good reason why assigning a vspace to an asid pool doesn't require you to specify the slot index to use? 

Probably not, tbh. The topic has come up in the past in [SELFOUR-2363](https://sel4.atlassian.net/browse/SELFOUR-2364) and [SELFOUR-2364](https://sel4.atlassian.net/browse/SELFOUR-2363)

> The kernel can spend a long time searching for a free asid slot if it's always trying to find the next free one. And it's easy to reuse ASIDs unintentionally if you revoke an address space and then create a new one, it'll use the same slot I believe.

Yes, agreed. I don't think it would be terribly hard to verify that solution, but it would probably break a lot of user-space code. Then again, you could implement the same policy in a library that simulates the old interface.